edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ServiceIdentityTree.cs

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Core
```

Comment:
Is it possible to create cycles by mistake in the configuration? If that's the case, code might crash in recursively looping

Reply:
Yes, we should definitely handle cycles